firmware/Core/Src/adcs_drivers/adcs_commands.c

```diff
@@ -0,0 +1,1035 @@
+
+	ADCS_Pack_to_Program_Status_Struct(&data_received[0], output_struct);
+
+	return tlm_status;
+}
+
+/// @brief Instruct the ADCS to execute the ADCS_Communication_Status command.
+/// @param output_struct Pointer to struct in which to place packed ADCS telemetry data
+/// @return 0 if successful, non-zero if a HAL or ADCS error occurred in transmission.
+uint8_t ADCS_Get_Communication_Status(ADCS_Comms_Status_Struct *output_struct) {
+	// returns I2C communication status of the ADCS (Table 37)
+	uint8_t data_length = 6;
+	uint8_t data_received[data_length]; // define temp buffer
+
+	uint8_t tlm_status = ADCS_I2C_telemetry_wrapper(ADCS_TELEMETRY_SATSTATE_COMM_STATUS, data_received, data_length, ADCS_INCLUDE_CHECKSUM);
+
+	ADCS_Pack_to_Comms_Status_Struct(&data_received[0], output_struct);
+
+	return tlm_status;
+}
+
+/// @brief Instruct the ADCS to execute the ADCS_Deploy_Magnetometer command.
+/// @return 0 if successful, non-zero if a HAL or ADCS error occurred in transmission.
+uint8_t ADCS_Deploy_Magnetometer(uint8_t deploy_timeout) {
+	// Deploys the magnetometer boom, timeout in seconds
+	uint8_t data_send[1] = {deploy_timeout};
+	uint8_t cmd_status = ADCS_I2C_telecommand_wrapper(ADCS_COMMAND_DEPLOY_MAGNETOMETER_BOOM, data_send, sizeof(data_send), ADCS_INCLUDE_CHECKSUM);
+	return cmd_status;
+}
+
+/// @brief Instruct the ADCS to execute the ADCS_Set_Run_Mode command.
+/// @return 0 if successful, non-zero if a HAL or ADCS error occurred in transmission.
+uint8_t ADCS_Set_Run_Mode(ADCS_Run_Mode mode) {
+	// Disables the ADCS
+	uint8_t data_send[1] = {mode};
+	uint8_t cmd_status = ADCS_I2C_telecommand_wrapper(ADCS_COMMAND_ADCS_RUN_MODE, data_send, sizeof(data_send), ADCS_INCLUDE_CHECKSUM);
+	return cmd_status;
+}
+
+/// @brief Instruct the ADCS to execute the ADCS_Clear_Errors command.
+/// @return 0 if successful, non-zero if a HAL or ADCS error occurred in transmission.
+uint8_t ADCS_Clear_Errors() {
+	// Clears error flags
+	// NOTE: THERE IS ANOTHER, SEPARATE CLEAR ERROR FLAG TC FOR THE BOOTLODER (ADCS_COMMAND_BL_CLEAR_ERRORS)
+	uint8_t data_send[1] = {192}; // 0b11000000
+	uint8_t cmd_status = ADCS_I2C_telecommand_wrapper(ADCS_COMMAND_CLEAR_ERRORS, data_send, sizeof(data_send), ADCS_INCLUDE_CHECKSUM);
+	return cmd_status;
+}
+
+/// @brief Instruct the ADCS to execute the titular command.
+/// @param mode Control mode to set (Table 77 in Firmware Manual)
+/// @param timeout Timeout to set control mode
+/// @return 0 if successful, non-zero if an error occurred in transmission.
+uint8_t ADCS_Attitude_Control_Mode(ADCS_Control_Mode mode, uint16_t timeout) {
+	// Sets the ADCS attitude control mode
+	// See User Manual, Section 4.4.3 Table 3 for requirements to switch control mode
+	uint8_t data_send[3] = {mode, timeout & 0x00FF, timeout >> 8};
+	uint8_t cmd_status = ADCS_I2C_telecommand_wrapper(ADCS_COMMAND_SET_ATTITUDE_CONTROL_MODE, data_send, sizeof(data_send), ADCS_INCLUDE_CHECKSUM);
```

Comment:
Please rename `ADCS_I2C_telecommand_wrapper` to describe what it does. Perhaps `ADCS_I2C_send_cmd_get_response`?

Reply:
This is done. I went with `ADCS_i2c_send_command_and_check`.